modules/oe_content_person/oe_content_person.module

```diff
@@ -42,3 +44,77 @@ function oe_content_person_node_presave(EntityInterface $entity) {
     }
   }
 }
+
+/**
+ * Implements hook_inline_entity_form_entity_form_alter().
+ */
+function oe_content_person_inline_entity_form_entity_form_alter(array &$entity_form, FormStateInterface &$form_state) {
+  if ($entity_form['#entity_type'] !== 'oe_person_job') {
+    return;
+  }
+
+  // Set initial state when form is loaded. If field
+  // "What type of person are you adding?" is not set then all fields will
+  // be hidden.
+  $type = NULL;
+  $user_input = $form_state->getUserInput();
+  if (isset($user_input['oe_person_type'])) {
+    $type = $user_input['oe_person_type'];
+  }
+  if ($type !== 'eu') {
+    $entity_form['oe_role_reference']['#attributes']['style'] = 'display:none';
+    $entity_form['oe_acting']['#attributes']['style'] = 'display:none';
+    $entity_form['oe_role_name']['widget'][0]['value']['#required'] = TRUE;
+  }
+
+  if ($type !== 'non_eu') {
+    $entity_form['oe_role_name']['#attributes']['style'] = 'display:none';
+    $entity_form['oe_role_reference']['widget'][0]['target_id']['#required'] = TRUE;
+  }
+
+  // Set states based on "What type of person are you adding?" field.
+  ContentFormUtilities::toggleFieldByValue($entity_form, 'oe_person_type', 'oe_role_name', 'non_eu');
+  ContentFormUtilities::toggleFieldByValue($entity_form, 'oe_person_type', 'oe_role_reference', 'eu');
+  ContentFormUtilities::toggleFieldByValue($entity_form, 'oe_person_type', 'oe_acting', 'eu');
+
+  $entity_form['oe_role_name']['widget'][0]['value']['#states']['required'] = [
+    ':input[name="oe_person_type"]' => [
+      'value' => 'non_eu',
+    ],
+  ];
+  $entity_form['oe_role_reference']['widget'][0]['target_id']['#states']['required'] = [
+    ':input[name="oe_person_type"]' => [
+      'value' => 'eu',
+    ],
+  ];
+}
+
+/**
+ * Validation callback for a "oe_person_jobs" element.
+ *
+ * @param array $element
+ *   The element being processed.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ * @param array $complete_form
+ *   The complete form structure.
+ */
+function oe_content_person_oe_person_jobs_validate(array &$element, FormStateInterface $form_state, array &$complete_form): void {
```

Comment:
We can't use oe_content_person_oe_person_job_presave() here because main entity is saved after Job is saved. Therefore old type is available here only.

We can use oe_content_person_node_presave() and load all referenced jobs with cleaning theirs fields. But it means that on one Person update we will save Jobs twice. Let's use form validation for now.

Reply:
was also thinking in the parent entities presave, I think this needs to be discussed at team level

Reply:
it was discussed with @ademarco to leave it as is.